### PR TITLE
nspawn, vmspawn, run0: add env var for turning off background tinting

### PR DESCRIPTION
**Pull request by poettering**
Some people are just sad, sad lost souls who don't like even the tiniest ray of color in their life. Let's add an env var knob for allowing them to turn the background tinting off, to drive the last bit of color from their life so that they can stay in their grey grey life.

### CODE REVIEW

File: docs/ENVIRONMENT.md [hunk 1]
```diff
@@ -705,3 +705,11 @@ Tools using the Varlink protocol (such as `varlinkctl`) or sd-bus (such as
   placed in a trusted disk image directory (see above), or if suitable polkit
   authentication was acquired. See `systemd.image-policy(7)` for the valid
   syntax for image policy strings.
+
+`systemd-run`, `run0`, `systemd-nspawn`, `systemd-vmspawn`:
+
+* `$SYSTEMD_TINT_BACKGROUND` – Takes a boolean. When false the automatic
+  tinting of the background for containers, VMs, and interactive `systemd-run`
+  and `run0` invocaitons is turned off. Not that this environment variable has
```
Comment (YHNdnzj) [inline docs/ENVIRONMENT.md]:
```suggestion
  and `run0` invocations is turned off. Not that this environment variable has
```

**Comment by poettering** [comment]
> Hmm, isn't disabling this already possible with `--background=""`? I don't see the point?

Yes, it's pretty much the same thing. The only reason to support an env var for this is to make it easy to turn this off in all tools at once.

If people want to turn this off via --background= globally they#d have to add at least four shell aliases, and possibly more in future. And env var is easier to manage, it's a single knob for all tools, and works independently of shells.

File: src/shared/pretty-print.c [hunk 2]
```diff
@@ -444,6 +444,19 @@ int terminal_tint_color(double hue, char **ret) {
...
 }
 
+bool shall_tint_background(void) {
+        static int cache = -1;
+
+        if (cache >= 0)
+                return cache;
+
+        cache = getenv_bool("SYSTEMD_TINT_BACKGROUND");
+        if (cache < 0 && cache != -ENXIO)
```
Comment (YHNdnzj) [inline src/shared/pretty-print.c]: When ENXIO, false should be cached?

```c
if (cache == -ENXIO)
        return (cache = false);
if (cache < 0)
        ...

return ...
```

**Review by YHNdnzj — APPROVED** [review]

File: docs/ENVIRONMENT.md [hunk 3]
```diff
@@ -705,3 +705,11 @@ Tools using the Varlink protocol (such as `varlinkctl`) or sd-bus (such as
   placed in a trusted disk image directory (see above), or if suitable polkit
   authentication was acquired. See `systemd.image-policy(7)` for the valid
   syntax for image policy strings.
+
+`systemd-run`, `run0`, `systemd-nspawn`, `systemd-vmspawn`:
+
+* `$SYSTEMD_TINT_BACKGROUND` – Takes a boolean. When false the automatic
+  tinting of the background for containers, VMs, and interactive `systemd-run`
+  and `run0` invocations is turned off. Not that this environment variable has
```
Comment (mrc0mmand) [inline docs/ENVIRONMENT.md]:
```suggestion
  and `run0` invocations is turned off. Note that this environment variable has
```

**Review by bluca — APPROVED** [review]

**Pull request merged**